HDFS-17854 Fix for Namenode Web UI file deletion bug

Jimvin opened the pull request:
### Description of PR
This is a fix for bug described in HDFS-17854 that changes the behaviour of the namenode web UI delete modal to prevent unexpected file deletion.

### How was this patch tested?
The patch was tested manually to confirm the expected behaviour is observed when deleting files in the web UI.

### For code changes:

- [x] Does the title or this PR starts with the corresponding JIRA issue id (e.g. 'HADOOP-17799. Your PR title ...')?
- [ ] Object storage: have the integration tests been executed and the endpoint declared according to the connector-specific documentation?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the `LICENSE`, `LICENSE-binary`, `NOTICE-binary` files?

Jimvin committed:
Added documentation for dfs.datanode.synconclose to hdfs-default.xml

Jimvin committed:
Merge pull request #1 from Jimvin/synconclose-docs-patch

HDFS-17794 Documentation for dfs.datanode.synconclose

Jimvin committed:
Merge branch 'apache:trunk' into trunk

Commit:
HDFS-17854 Fix for Namenode Web UI file deletion bug

Commit:
Revert "HDFS-17854 Fix for Namenode Web UI file deletion bug"

This reverts commit 56b344e51b327c7794fac298fe6f17dc87ffaa58.

Commit:
HDFS-17854 Fix for Namenode Web UI file deletion bug

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 47s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +0 :ok: |  xmllint  |   0m  0s |  |  xmllint was not available.  |
| +0 :ok: |  jshint  |   0m  0s |  |  jshint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
|||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  34m 11s |  |  trunk passed  |
| +1 :green_heart: |  compile  |   1m 20s |  |  trunk passed with JDK Ubuntu-21.0.7+6-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  compile  |   1m 25s |  |  trunk passed with JDK Ubuntu-17.0.15+6-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  mvnsite  |   1m 36s |  |  trunk passed  |
| +1 :green_heart: |  javadoc  |   1m 11s |  |  trunk passed with JDK Ubuntu-21.0.7+6-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   1m  9s |  |  trunk passed with JDK Ubuntu-17.0.15+6-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  shadedclient  |  66m 39s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   1m 21s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 21s |  |  the patch passed with JDK Ubuntu-21.0.7+6-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javac  |   1m 21s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 20s |  |  the patch passed with JDK Ubuntu-17.0.15+6-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javac  |   1m 20s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  mvnsite  |   1m 28s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   1m  1s |  |  the patch passed with JDK Ubuntu-21.0.7+6-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   1m  4s |  |  the patch passed with JDK Ubuntu-17.0.15+6-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  shadedclient  |  29m 33s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| -1 :x: |  unit  | 213m 11s | [/patch-unit-hadoop-hdfs-project_hadoop-hdfs.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-8068/1/artifact/out/patch-unit-hadoop-hdfs-project_hadoop-hdfs.txt) |  hadoop-hdfs in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 47s |  |  The patch does not generate ASF License warnings.  |
|  |   | 318m  2s |  |  |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.tools.TestHdfsConfigFields |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.51 ServerAPI=1.51 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-8068/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/8068 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient codespell detsecrets xmllint jshint |
| uname | Linux fc962a5104ea 5.15.0-160-generic #170-Ubuntu SMP Wed Oct 1 10:06:56 UTC 2025 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 1c0eaf2cd0a506bc893f2f8c03c413d3eadacace |
| Default Java | Ubuntu-17.0.15+6-Ubuntu-0ubuntu120.04 |
| Multi-JDK versions | /usr/lib/jvm/java-21-openjdk-amd64:Ubuntu-21.0.7+6-Ubuntu-0ubuntu120.04 /usr/lib/jvm/java-17-openjdk-amd64:Ubuntu-17.0.15+6-Ubuntu-0ubuntu120.04 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-8068/1/testReport/ |
| Max. process+thread count | 3600 (vs. ulimit of 5500) |
| modules | C: hadoop-hdfs-project/hadoop-hdfs U: hadoop-hdfs-project/hadoop-hdfs |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-8068/1/console |
| versions | git=2.25.1 maven=3.9.11 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

Jimvin commented:
Manual testing was performed on the namenode webUI to confirm that this change addresses the unexpected file deletion issue.

The following steps can be used to reproduce this issue before this change was applied:
- Open the namenode web UI and select utilities -> Browse the file system
- Click on the trashcan delete icon next to a file
- Click the cancel button on the delete confirmation modal
- Click in the delete icon for a different file
- Click the Delete button on the delete confirmation modal
- Both files are deleted

After applying the change the documented behaviour from manual testing:
- Open the namenode web UI and select utilities -> Browse the file system
- Click on the trashcan delete icon next to a file
- Click the cancel button on the delete confirmation modal
- Click in the delete icon for a different file
- Click the Delete button on the delete confirmation modal
- Only the last selected file was deleted

slfan1989 commented:
@Jimvin Thank you for your contribution, but I have a question: what is the relationship between this and adding a configuration?

Jimvin committed:
Remove dfs.datanode.synconclose property

Removed the dfs.datanode.synconclose property and its description from hdfs-default.xml.

Jimvin commented:
The configuration change was checked in in error, I have removed it from the PR.

slfan1989 commented:
@Jimvin Could we create a new Jira for this PR?

Jimvin commented:
@slfan1989 I opened HDFS-17854 specifically for this fix, do you still want a new Jira created? I also linked a couple of older issues that appear to be duplicates to HDFS-17854.

ayushtkn reviewed (approved):
LGTM
